packages/components/src/components/forms/input/DatePickerInput.tsx

```diff
@@ -0,0 +1,192 @@
+import DatePicker from 'react-datepicker';
+
+import { Utils } from '@labkey/api';
+
+import {DisableableInput, DisableableInputProps, DisableableInputState} from "./DisableableInput";
+import { FieldLabel } from "../FieldLabel";
+import { QueryColumn } from '../../base/models/model';
+import {datePlaceholder, formatDate, isDateTimeCol, parseDate} from '../../../util/Date';
+
+export interface DatePickerInputProps extends DisableableInputProps {
+    formsy?: boolean
+    inputClassName?: string
+    inputWrapperClassName?:string
+    disabled?: boolean
+    isClearable?: boolean
+    placeholderText?: string
+    name?: string
+    label?: any
+    onChange?: any
+    dateFormat?: string
+    showTime?: boolean
+
+    queryColumn: QueryColumn
+    showLabel?: boolean
+    value?: any
+    addLabelAsterisk?: boolean
+
+    // from formsy-react
+    getErrorMessage?: Function
+    getValue?: Function
+    setValue?: Function
+    showRequired?: Function
+    validations?: any
+}
+
+interface DatePickerInputState extends DisableableInputState {
+    selectedDate: any
+    selectedDateStr: string
+}
+
+class DatePickerInputImpl extends DisableableInput<DatePickerInputProps, DatePickerInputState> {
+
+    static defaultProps = {
+        allowDisable: false,
+        initiallyDisabled: false,
+        isClearable: true,
+        inputClassName: 'form-control',
+        inputWrapperClassName: 'block',
+        showLabel: true,
+        addLabelAsterisk: false
+    };
+
+    constructor(props: DatePickerInputProps) {
+        super(props);
+
+        this.onChange = this.onChange.bind(this);
+        this.toggleDisabled = this.toggleDisabled.bind(this);
+
+        this.state = {
+            isDisabled: props.initiallyDisabled,
+            selectedDate: props.value ? parseDate(props.value, this.getDateFormat()) : undefined,
+            selectedDateStr: props.value
+        }
+    }
+
+    onChange(date) {
+        const selectedDateStr = date ? formatDate(date, null, this.getDateFormat()): undefined;
+        this.setState(() => {
+            return {
+                selectedDate: date,
+                selectedDateStr
+            }
+        });
+
+        if (this.props.onChange && Utils.isFunction(this.props.onChange))
+            this.props.onChange(date);
+
+        if (this.props.formsy && Utils.isFunction(this.props.setValue))
+            this.props.setValue(selectedDateStr);
+    }
+
+    getDateFormat() {
+        const { dateFormat, queryColumn } = this.props;
+        if (dateFormat)
+            return this.ensureDateFormat(dateFormat);
+
+        return this.ensureDateFormat(datePlaceholder(queryColumn));
+    }
+
+    ensureDateFormat(rawFormat: string) {
+        return rawFormat.replace('YYYY', 'yyyy').replace('DD', 'dd');
+    }
+
+    shouldShowTime() {
+        const { showTime, queryColumn } = this.props;
+        return showTime || isDateTimeCol(queryColumn)
+    }
+
+    render() {
```

Comment:
The layout of this element is really off when the "allowDisable" property is true.

Reply:
fixed